infoschema: `TestSlowQueryOOM` fails in some timezones

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Change system timezone to 'UTC-8' and run `TestSlowQueryOOM`

### 2. What did you expect to see? (Required)

Test succeeded.

### 3. What did you see instead (Required)

Test failed:

```
--- FAIL: TestSlowQueryOOM (1.05s)
    cluster_tables_test.go:721:
        	Error Trace:	/Users/bb7133/tidb/pkg/infoschema/test/clustertablestest/cluster_tables_test.go:721
        	            				/Users/bb7133/tidb/pkg/infoschema/test/clustertablestest/cluster_tables_test.go:726
        	Error:      	An error is expected but got nil.
        	Test:       	TestSlowQueryOOM
        	Messages:   	2048
```

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->
7e51c7732c569e187ba6d0e4c3bd1c46320026fc

### 5. Inspection

The test failed since timezone specified in the [test slow log('tidb-slow-*.log') is 'UTF+8'](https://github.com/pingcap/tidb/blob/7e51c7732c569e187ba6d0e4c3bd1c46320026fc/pkg/infoschema/test/clustertablestest/cluster_tables_test.go#L659):

```
# Time: 2022-04-14T10:50:28.185954+08:00
# Txn_start_ts: 432512598850928660
...
```

But the [query of slow logs](https://github.com/pingcap/tidb/blob/7e51c7732c569e187ba6d0e4c3bd1c46320026fc/pkg/infoschema/test/clustertablestest/cluster_tables_test.go#L720) uses system timezone, in some timezones like 'UTC-8', the query condition hits nothing and consumes little memory, in such timezones the query will not fail on expectation.

--- a/pkg/infoschema/test/clustertablestest/cluster_tables_test.go
+++ b/pkg/infoschema/test/clustertablestest/cluster_tables_test.go
@@ -714,6 +714,8 @@ select * from t1;
 	tk.MustExec("set global tidb_mem_oom_action='CANCEL'")
 	defer tk.MustExec("set global tidb_mem_oom_action='LOG'")
 	tk.MustExec(fmt.Sprintf("set @@tidb_slow_query_file='%v'", f.Name()))
+	// Align with the timezone in slow log files
+	tk.MustExec("set @@time_zone='+08:00'")
 	checkFn := func(quota int) {
 		tk.MustExec("set tidb_mem_quota_query=" + strconv.Itoa(quota)) // session
 

